com.unity.hlod/Runtime/HLODcameraRecognizerManager.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.HLODSystem
+{
+    public class HLODCameraRecognizerManager
+    {
+        private static HLODCameraRecognizerManager s_instance; 
+        public static HLODCameraRecognizerManager Instance
+        {
+            get
+            {
+                if (s_instance == null)
+                    s_instance = new HLODCameraRecognizerManager();
+                return s_instance;
+            }
+        }
+
+        public static HLODCameraRecognizer ActiveRecognizer
+        {
+            get
+            {
+                var instance = Instance;
+                if (instance.m_activeRecognizer == null)
+                {
+                    instance.ActiveHighestPriority();
+                }
+
+                return Instance.m_activeRecognizer;
+            }
+        }
+        public static Camera ActiveCamera
+        {
+            get
+            {
+                var instance = Instance;
+                var regocnizer = ActiveRecognizer;
+                if (regocnizer == null)
+                    return null;
+
+                return ActiveRecognizer.RecognizedCamera;
+            }
+        }
+
+        private bool m_enableAutoHighestPrioritySetting = true;
+        private HLODCameraRecognizer m_activeRecognizer = null;
+        private List<HLODCameraRecognizer> m_recognizers = new List<HLODCameraRecognizer>();
+        
+        public void RegisterRecognizer(HLODCameraRecognizer recognizer)
+        {
+            m_recognizers.Add(recognizer);
+            m_recognizers.Sort((lhs, rhs) =>
+            {
+                //sort in descending order
+                return rhs.Priority - lhs.Priority;
```

Comment:
This is more or less a question rather than a review point, but is there an advantage to use subtraction in comparison to 'CompareTo()'?? :)

Reply:
To compare the two values, we simply used the subtraction operation. I just prefer this method than CompareTo.

Reply:
Understood!! and it makes perfect sense!!!